### PR TITLE
feat: implementing payload to call createFunction function

### PR DESCRIPTION
**Pull request by temtechie**

### CODE REVIEW

File: src/app/api/loan/route.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { NextResponse, NextRequest } from "next/server";
+import connectDatabase from "@/lib/database";
+import LoanModel from "@/models/loan-model";
+
+export async function POST(request: NextRequest) {
+
+    const body = await request.json();
+    if (!body.userAddress) {
+        return NextResponse.json({ message: "User wallet address is required" }, { status: 400 });
+    }
```
Comment (abelbordonado) [inline src/app/api/loan/route.ts:10]: This validation is far not enough for an API

Lets use express validator, you need to validate each parameter of the JSON (which I think they are several)

https://yaali4422519.medium.com/next-js-validate-request-using-next-connect-and-express-validator-464997067a9c

File: src/app/component/create-loan/create-loan-form.tsx [hunk 2]
```diff
@@ -42,10 +84,41 @@ const CreateLoanForm = () => {
...
+      args: [
+        address,
+        formValues?.loanAmount,
+        formValues?.loanAmountToken,
+        formValues?.collateralAmount,
+        formValues?.collateralAmountToken,
+        formValues?.loanPeriod,
+        formValues?.interestRate,
+      ],
+    });
```
Comment (abelbordonado) [inline src/app/component/create-loan/create-loan-form.tsx:117]: We don't call to any web3 or API directly from the section components.

Lets wrap this write contract into their own layer

File: src/app/component/create-loan/select-collateral-token.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import React from "react";
+
+type SelectCollateralTokenProps = {
+  options: { name: string; tokenAddress: any }[];
```
Comment (abelbordonado) [inline src/app/component/create-loan/select-collateral-token.tsx:4]: tokenaAddress: string

File: src/data/create-loan-position.ts [hunk 4]
```diff
@@ -1,7 +1,12 @@
 export const allowedCoins = [
-    { value: "USDC", icon: "...", token: "USDC" },
-    { value: "USD", icon: "...", token: "USD" },
-    { value: "LINK", icon: "...", token: "LINK" },
+    { name: "USDC", tokenAddress: '0x51fCe89b9f6D4c530698f181167043e1bB4abf89' },
+    { name: "wBTC", tokenAddress: '0xFF82bB6DB46Ad45F017e2Dfb478102C7671B13b3' },
```
Comment (abelbordonado) [inline src/data/create-loan-position.ts:3]: You understood this is good 👍

File: src/interfaces/loan-interface.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
+export interface ILoan {
```
Comment (abelbordonado) [inline src/interfaces/loan-interface.ts:1]: Personal Opinion, I m not fan of the `I` in front just to say is Interface, can be just `Loan`

File: src/models/loan-model.ts [hunk 6]
```diff
@@ -0,0 +1,18 @@
+import { ILoan } from "@/interfaces/loan-interface";
+import { Document, Schema, model } from 'mongoose';
+
+const loanSchema = new Schema<ILoan>({
```
Comment (abelbordonado) [inline src/models/loan-model.ts:4]: Add here the LoanManager address used to create this contract

File: src/services/api/loan-service.ts [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+                'Content-Type': 'application/json',
+            },
+            body: JSON.stringify(loanData),
+        });
+        const newLoan = await response.json();
+        return newLoan;
+    } catch (error) {
+        console.log("Failed creating new loan, api/loan: ", error);
+
+    }
```
Comment (abelbordonado) [inline src/services/api/loan-service.ts:18]: Return something in case of loan, if is an object return a null

**Comment by abelbordonado** [comment]
image

**Comment by abelbordonado** [comment]
image

File: src/app/api/loan/route.ts (hunk 1 above)
Comment (temtechie) [reply src/app/api/loan/route.ts:10]: it was just a placeholder  though, not fully implemented

File: src/app/component/create-loan/create-loan-form.tsx (hunk 2 above)
Comment (temtechie) [reply src/app/component/create-loan/create-loan-form.tsx:117]: noted

File: src/app/component/create-loan/select-collateral-token.tsx (hunk 3 above)
Comment (temtechie) [reply src/app/component/create-loan/select-collateral-token.tsx:4]: noted

File: src/interfaces/loan-interface.ts (hunk 5 above)
Comment (temtechie) [reply src/interfaces/loan-interface.ts:1]: do you have any special reason?  I could learn from you

File: src/services/api/loan-service.ts (hunk 7 above)
Comment (temtechie) [reply src/services/api/loan-service.ts:18]: noted

**Pull request merged**